homeassistant/components/yolink/select.py

```diff
@@ -0,0 +1,117 @@
+        translation_key="sprinkler_mode",
+        icon="mdi:auto-mode",
+        value=lambda data: data.get("mode")
+        if data is not None
+        else None,  # watering state report will missing state field
+        exists_fn=lambda device: device.device_type == ATTR_DEVICE_SPRINKLER,
+        should_update_entity=lambda value: value is not None,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up YoLink select from a config entry."""
+    device_coordinators = hass.data[DOMAIN][config_entry.entry_id].device_coordinators
+    selector_device_coordinators = [
+        device_coordinator
+        for device_coordinator in device_coordinators.values()
+        if device_coordinator.device.device_type in [ATTR_DEVICE_SPRINKLER]
+    ]
+    async_add_entities(
+        YoLinkSelectEntity(config_entry, selector_device_coordinator, description)
+        for selector_device_coordinator in selector_device_coordinators
+        for description in SELECTOR_MAPPINGS
+        if description.exists_fn(selector_device_coordinator.device)
+    )
+
+
+class YoLinkSelectEntity(YoLinkEntity, SelectEntity):
+    """YoLink Select Entity."""
+
+    entity_description: YoLinkSelectEntityDescription
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: YoLinkCoordinator,
+        description: YoLinkSelectEntityDescription,
+    ) -> None:
+        """Init YoLink Select."""
+        super().__init__(config_entry, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = (
+            f"{coordinator.device.device_id} {self.entity_description.key}"
+        )
+
+    @callback
+    def update_entity_state(self, state: dict[str, Any]) -> None:
+        """Update HA Entity State."""
+        if (
+            current_value := self.entity_description.value(
+                state.get(self.entity_description.state_key)
+            )
+        ) is None and self.entity_description.should_update_entity(
+            current_value
+        ) is False:
+            return
+        self._attr_current_option = current_value
+        self.async_write_ha_state()
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected option."""
+        device_type = self.coordinator.device.device_type
+        if device_type == ATTR_DEVICE_SPRINKLER:
+            data = await self.call_device(
+                ClientRequest(
+                    "setState",
+                    {
+                        "state": {
+                            "mode": option,
+                        }
+                    },
+                )
+            )
+            self._attr_current_option = option
+            sprinkler_message_resolve(self.coordinator.device, data, None)
+            self.coordinator.async_set_updated_data(data)
```

Comment:
How will future select entities work? Will they all change a mode? Because now we make the select entity less generic

Reply:
Used to configure device parameters

Reply:
Let me rephrase, will future select entities also be using `setState` like this?